src/Evolution/Particles/MonteCarlo/Actions/InitializeMonteCarlo.hpp

```diff
@@ -132,26 +139,38 @@ struct InitializeMCTags {
         tmpl::list<Particles::MonteCarlo::Tags::RandomNumberGenerator>>(
         make_not_null(&box), std::move(rng));
 
-    // Currently hard-code energy at emission; should be set by option
+    // Initial energy of packets, read from MC options
     typename Particles::MonteCarlo::Tags::DesiredPacketEnergyAtEmission<
         NeutrinoSpecies>::type packet_energy_at_emission =
         make_with_value<std::array<DataVector, NeutrinoSpecies>>(
-            DataVector{num_grid_points}, 1.e-12);
+            DataVector{num_grid_points}, 0.0);
+    for (size_t s = 0; s < NeutrinoSpecies; s++) {
+      packet_energy_at_emission[s] = initial_packet_energy[s];
+    }
     Initialization::mutate_assign<
         tmpl::list<Particles::MonteCarlo::Tags::DesiredPacketEnergyAtEmission<
             NeutrinoSpecies>>>(make_not_null(&box),
                                std::move(packet_energy_at_emission));
 
-    typename Particles::MonteCarlo::Tags::CellLightCrossingTime<
-        DataVector>::type cell_light_crossing_time(num_grid_points, 1.0);
-    Initialization::mutate_assign<tmpl::list<
-        Particles::MonteCarlo::Tags::CellLightCrossingTime<DataVector>>>(
-        make_not_null(&box), std::move(cell_light_crossing_time));
-
-    // Initialize empty mortar data; do we need more at initialization stage?
+    // Initialize mortar data. Currently assumes a single neighbor on each
+    // face (i.e. no h-refinement)
     using MortarData =
         typename Particles::MonteCarlo::Tags::MortarDataTag<dim>::type;
     MortarData mortar_data;
+    const Element<dim>& element = db::get<::domain::Tags::Element<dim>>(box);
```

Comment:
Do you need to do this initialization?  please check

Reply:
@kidder Do you mean the element, or the entire mortar_data? I initialize the mortar_data largely to have buffers of the correct size preassigned. 
If that's a concern, it would likely be possible to get the same information on the fly from the mesh around L329 of 
`src/Evolution/Particles/MonteCarlo/GhostZoneCommunication.hpp`, at least as long as we do not have mesh refinement. Overall the idea was to avoid figuring out every time what the size of the communication buffer / ghost zone region should be and just use reserved space for the GZ data.

Reply:
the entire mortar_data;  It is not a concern, there are two possibe ways to do this: 1) you need to initialize here with the correct sizes and there is at most an ASSERT that checks they are the correct size; 2) you default initialize here, and whatever uses it first checks the size and resizes if necessary.  The first is fine if you don't expect to p-refine (and currently we don't plan to do so for regions using subcell).